wave_gtsam/include/wave/gtsam/preint_imu_factor.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef WAVE_PREINT_IMU_FACTOR_HPP
+#define WAVE_PREINT_IMU_FACTOR_HPP
+
+#include <gtsam/nonlinear/NonlinearFactor.h>
+#include <gtsam/base/Matrix.h>
+#include "wave/gtsam/pose_vel_bias.hpp"
+#include "wave/gtsam/pose_vel.hpp"
+#include <gtsam/navigation/CombinedImuFactor.h>
+#include <gtsam/navigation/NavState.h>
+
+namespace wave {
+
+/**
+ * 4-ary factor connecting two combined pose/vel/bias states and two IMU biases.
+ * Provides the effect of gtsam::CombinedImuFactor (a 6-ary factor).
+ *
+ * @tparam StateType either PoseVel or PoseVelBias
+ */
+template <typename StateType>
+class PreintegratedImuFactor
+  : public gtsam::NoiseModelFactor4<StateType,
+                                    StateType,
+                                    gtsam::imuBias::ConstantBias,
+                                    gtsam::imuBias::ConstantBias> {
+ private:
+    gtsam::PreintegratedCombinedMeasurements pim;
+    using Base = gtsam::NoiseModelFactor4<StateType,
```

Comment:
just curious, why is this `using` statement private? Not sure if I've seen that before

Reply:
That means the alias is just for convenience of writing the class definition. It is used in GTSAM, presumably that's why Kinjal put it in. (I changed from `typedef` to `using`.)

Reply:
Cool.